docs/time-varying-infectiousness.md

```diff
@@ -0,0 +1,219 @@
+    - We also require that the generation interval have an inverse CDF or we be able to approximate it (i.e.,
+from empirical data).
+2. Infectiousness over time is a relative quantity, so it is completely separate from the absolute degree
+to which someone is infectious, represented by the total number of secondary infection attempts they have.
+That quantity, denoted $C_i$ in the generality, is directly related to $R_0$.
+3. We require $C_i$ ordered draws from the generation interval distribution. We schedule infection attempts
+at these times.
+
+## Central claim
+
+We could just take $C_i$ draws from the generation interval (GI) and order them, scheduling an infection
+attempt at each time drawn. However, we argue that we instead want to draw the minimum of our $C_i$ draws,
+have an infection attempt, and _then_ schedule the next infection attempt -- drawing the second
+smallest of our $C_i$ ordered draws from the GI. We argue that we want to use this approach because (a) it
+provides a great deal of flexibility, allowing for many changes to the GI, and (b) because it is more
+computationally efficient than alternatives.
+
+However, wanting to sequentially schedule the infection attempts requires that we have a way of getting
+the smallest of $C_i$ draws from the GI, the second smallest, the third smallest, etc. This is the problem
+of obtaining ordered draws from a distribution, or having information about the distribution's
+"[order statistics](https://en.wikipedia.org/wiki/Order_statistic)", and it is readily solvable for the
+case of the uniform distribution.
+
+### Why do we need order statistics?
+
+Let us first explore why scheduling the next infection attempt after the last (i.e., sequentially
+drawing infection attempt times from the GI) is the better approach compared to taking all the draws
+from the GI at the beginning of an individual's infectiousness period and at once scheduling the infection
+attempts based on those times.
+
+1. **Changes in the number of infection attempts in the middle of an agent's infectious course.** Imagine
+an agent dies while they are still infectious. Clearly, they cannot be infecting others. (Or, if the disease
+in question is Ebola, there are well-defined processes by which they may still be infectious and those processes
+should not be lumped with the infection generation interval).
+
+    To ensure a dead person does not infect others, we would like to remove the plans where they infect others.
+If we had pre-scheduled all infections, we would have had to store the plan IDs for all the infections in
+`HashMap<PersonId, Vec<PlanId>>`. Then, each time we had executed one of these plans, we would have to have
+removed it from the vector, so that the entry for each `PersonId` tells us the plans we have _left_ for a given
+person. Then, when the agent dies, we could iterate through the remaining plans and cancel them.
```

Comment:
I don't think that this changes your basic argument, but this isn't the only implementation choice.

Instead, you can store the *times* of the next infection, rather than have them as plans, and then just plan the next one. This removes the need to cancel, and if you keep them in a sorted list, then you also don't need to iterate over to plan the next one.

As for the storage, that depends very much on the data structure. For instance, Person Properties are stored as a HashMap of Vecs of PersonProperty, so you actually need an item for each person, whether they have scheduled plans or not, so the only real additional cost is the Vec of times itself

Reply:
Thanks! I had not thought of this approach and appreciate this insight for now revising this section to drive home the main points concisely.